SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp

```diff
@@ -120,42 +197,46 @@ bool GenerateDefaultLaunchConfig(const FString& LaunchConfigPath, const FSpatial
-			}
-			Writer->WriteArrayEnd();
-			Writer->WriteObjectStart(TEXT("snapshots"));
-				Writer->WriteValue(TEXT("snapshot_write_period_seconds"), LaunchConfigDescription.World.SnapshotWritePeriodSeconds);
-			Writer->WriteObjectEnd();
-		Writer->WriteObjectEnd(); // World section end
-		Writer->WriteObjectStart(TEXT("load_balancing")); // Load balancing section begin
-			Writer->WriteArrayStart("layer_configurations");
-			for (const FWorkerTypeLaunchSection& Worker : LaunchConfigDescription.ServerWorkers)
-			{
-				WriteLoadbalancingSection(Writer, Worker.WorkerTypeName, Worker.Columns, Worker.Rows, Worker.bManualWorkerConnectionOnly);
-			}
-			Writer->WriteArrayEnd();
+				Writer->WriteValue(TEXT("chunk_edge_length_meters"), LaunchConfigDescription.World.ChunkEdgeLengthMeters);
+				Writer->WriteArrayStart(TEXT("legacy_flags"));
+				for (auto& Flag : LaunchConfigDescription.World.LegacyFlags)
+				{
+					WriteFlagSection(Writer, Flag.Key, Flag.Value);
+				}
+				Writer->WriteArrayEnd();
+				Writer->WriteArrayStart(TEXT("legacy_javaparams"));
+					for (auto& Parameter : LaunchConfigDescription.World.LegacyJavaParams)
+					{
+						WriteFlagSection(Writer, Parameter.Key, Parameter.Value);
+					}
+				Writer->WriteArrayEnd();
+				Writer->WriteObjectStart(TEXT("snapshots"));
+					Writer->WriteValue(TEXT("snapshot_write_period_seconds"), LaunchConfigDescription.World.SnapshotWritePeriodSeconds);
+				Writer->WriteObjectEnd();
+			Writer->WriteObjectEnd(); // World section end
+			Writer->WriteObjectStart(TEXT("load_balancing")); // Load balancing section begin
+				Writer->WriteArrayStart("layer_configurations");
+				for (const auto& Worker : InWorkers)
+				{
+					if (Worker.Value.WorkerLoadBalancing == nullptr)
+					{
+						UE_LOG(LogSpatialGDKDefaultLaunchConfigGenerator, Error, TEXT("Worker type '%s' is missing a load balancing setting."), *Worker.Key.ToString());
+						return false;
+					}
+					WriteLoadbalancingSection(Writer, Worker.Key, *Worker.Value.WorkerLoadBalancing, Worker.Value.bManualWorkerConnectionOnly);
+				}
+				Writer->WriteArrayEnd();
 			Writer->WriteObjectEnd(); // Load balancing section end
 			Writer->WriteArrayStart(TEXT("workers")); // Workers section begin
-			for (const FWorkerTypeLaunchSection& Worker : LaunchConfigDescription.ServerWorkers)
-			{
-				WriteWorkerSection(Writer, Worker);
-			}
-			// Write the client worker section
-			FWorkerTypeLaunchSection ClientWorker;
-			ClientWorker.WorkerTypeName = SpatialConstants::DefaultClientWorkerType;
-			ClientWorker.WorkerPermissions.bAllPermissions = true;
-			ClientWorker.bLoginRateLimitEnabled = false;
-			WriteWorkerSection(Writer, ClientWorker);
+				for (const auto& Worker : InWorkers)
+				{
+					WriteWorkerSection(Writer, Worker.Key, Worker.Value);
+				}
+				// Write the client worker section
+				FWorkerTypeLaunchSection ClientWorker;
+				ClientWorker.WorkerPermissions.bAllPermissions = true;
```

Comment:
`ClientWorker.WorkerTypeName = SpatialConstants::DefaultClientWorkerType;` is now missing, is this intended?

Reply:
I guess not :/

Reply:
Yes, WorkerTypeName field has been deprecated.